app/src/main/java/com/example/myapplication/FirstFragment.java

```diff
@@ -80,8 +80,29 @@ public void afterTextChanged(Editable s) {
                 if (binding.input.getText() == null){
                     return;
                 }
-                MainActivity.teamNumber = String.valueOf(binding.input.getText());
-                UserModel.getMatchData().setTeamNumber(Integer.parseInt(MainActivity.teamNumber));
+                String teamNumber = String.valueOf(binding.input.getText());
+                UserModel.getMatchData().setTeamNumber(Integer.parseInt(teamNumber));
+            }
+        });
+
+        binding.matchInput.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                if (binding.matchInput.getText() == null){
```

Comment:
`binding.matchInput.getText()` is called twice (here and 3 lines down) - consider storing it in a variable and reusing it.